### PR TITLE
[doc] Add resultslist schema

### PR DESCRIPTION
**Pull request by dcermak**
- add a schema for the `GET /build/<projectname>/_result` route
- correct `localbuild` -> `locallink`

### CODE REVIEW

**Comment by dcermak** [comment]
@mlschroe could you take a look? I think you're the best qualified to review this.

**Comment by hennevogel** [comment]
The schema is also failing in the CI. See https://app.circleci.com/pipelines/github/openSUSE/open-build-service/2099/workflows/f45fef5d-178b-4a8a-a33a-10453d47b67d/jobs/53605

**Comment by dcermak** [comment]
> The schema is also failing in the CI. See https://app.circleci.com/pipelines/github/openSUSE/open-build-service/2099/workflows/f45fef5d-178b-4a8a-a33a-10453d47b67d/jobs/53605

Next try as a rng schema to make @adrianschroeter happy ;-)

File: docs/api/api/jobstatus.rng [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        <element name="lastduration">
+          <a:documentation>
+            Time in seconds that the last build took.
+          </a:documentation>
+          <data type="nonNegativeInteger"/>
+        </element>
+      </optional>
+
+      <optional>
+        <element name="lastduration">
```
Comment (eduardoj) [inline docs/api/api/jobstatus.rng:48]: Isn't this optional element defined from line 47 to 54 repeated from line 38 to 45?

Comment (dcermak) [reply docs/api/api/jobstatus.rng:48]: In fact every attribute and element of `jobstatus` is optional: if no job is running at the moment, then you get just `<jobstatus/>` back from obs. I have added them around every optional element & attribute, as I was unsure if you can wrap them all in one optional.

Comment (dcermak) [reply docs/api/api/jobstatus.rng]: Ups, I misread your comment and the `lastduration` element was duplicated. Should be fixed now.

**Comment by dcermak** [comment]
Thanks @adrianschroeter!

**Pull request merged**